src/client.js

```diff
@@ -148,6 +155,11 @@ exports.createClient = function(reference, name) {
     // Remove possible trailing slash from baseUrl
     this._options.baseUrl = this._options.baseUrl.replace(/\/$/, '');
 
+    if (this._options.base) {
+      const urlPrefix = reference.name || reference.baseUrl.split('//')[1].split('.')[0];
```

Comment:
The second arm of this `||` is for compatibility, right?  If so, maybe better to separate it out for later deletion when it's no longer needed.